controllers/handler/nodenetworkconfigurationpolicy_controller.go

```diff
@@ -223,6 +224,18 @@ func (r *NodeNetworkConfigurationPolicyReconciler) Reconcile(_ context.Context,
 		policyconditions.Update(r.Client, r.APIClient, request.NamespacedName)
 	}
 
+	stats, err := nmstatectl.Statistic(enactmentInstance.Status.DesiredState)
+	if err != nil {
+		log.Error(err, "failed retrieving stats for desired state")
+	} else {
+		for _, t := range stats.Topology {
+			monitoring.ApplyTopologyTotal.WithLabelValues(t).Inc()
+		}
+		for _, f := range stats.Features {
+			monitoring.ApplyFeaturesTotal.WithLabelValues(f).Inc()
+		}
```

Comment:
Per https://github.com/nmstate/kubernetes-nmstate/pull/1221/files#r1431131839, as I understand, I'm afraid this will have unbound cardinality as it will contain network state information like hostnames/interfaces/address/ports, is that correct?

Reply:
> Per https://github.com/nmstate/kubernetes-nmstate/pull/1221/files#r1431131839, as I understand, I'm afraid this will have unbound cardinality as it will contain network state information like hostnames/interfaces/address/ports, is that correct?

The cardinality is not unbound, is not like we are going to count "mycluster1" and "eth1" the topology and is a conbination of enumerations (that's a little unbound but not that much) and the features is just an a list of enums

This can be a problem ?

Reply:
Yes, labels must have a limited values set. Except for specific labels like VM names, pod names, namespaces and other resources etc.
See note: https://prometheus.io/docs/practices/naming/#labels

Reply:
so this is an example of the nmstate stdats 
```
topology:
- static_ip4,static_ip6 -> linux-bridge -> bond -> ethernet
- static_ip4,static_ip6 -> vlan -> linux-bridge -> bond -> ethernet
features:
- sriov
- mac-based-identifier
```

For "features" is clear that is limited on the enum variants, maybe the main problem would be "topology".

@sradco @machadovilaca do you know what kind of limits do we have for labels bound ?